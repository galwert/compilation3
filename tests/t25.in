int func1(int param1)
{
    return 1;
}
int func2(int param1, int param2)
{
    return func1(param1) * param2;
}
int func3(int param1, int param2, bool param3)
{
    int param4 = func1(param1);
    return 5;
}
int func4()
{
    return 1;
}

void main()
{
    int x1 = func1(1);
    int x2 = func2(1,2);
    int x3 = func3(x1, x2, true);
    int x4 = func3(x1, x2, false);
    while (((false and false) and true) or true)
    {
        int y = 1;
        continue;
        break;
    }
    break; // Should fail here
}