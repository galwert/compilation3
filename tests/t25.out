---end scope---
param1 INT -1
---end scope---
param1 INT -1
param2 INT -2
---end scope---
param1 INT -1
param2 INT -2
param3 BOOL -3
param4 INT 0
---end scope---
---end scope---
y INT 4
---end scope---
line 31: unexpected break statement
